TurboHook Extension (Webhooks)

CubesterYT opened the pull request:
Late (But my first ever PR + Extension) pull request, was going to do it earlier, but had work to do.

TurboHook is a fully modular extension for Webhooks. It works with any program or site that uses a normal Webhook Syntax, such as Guilded and Discord. Currently, it can do normal webhooks, but I will add embeds in a later update, but this extension is done as of now.

Old
![image](https://github.com/TurboWarp/extensions/assets/78769806/afe8074b-8482-4264-af5f-6a322b681264)
Alternate Version (Which is now the currently used base)
![image](https://github.com/TurboWarp/extensions/assets/78769806/89b63535-0ef1-4739-983d-35b098de03bd)
Alternate Revamp (Current)
![image](https://github.com/TurboWarp/extensions/assets/78769806/bca6d434-0c17-40e2-a6c2-f546fee59ea3)

As per explaining how it works, you can find documentation <a href="https://github.com/CubesterYT/dependencies/blob/main/TurboWarp%20Extensions/TurboHook/TurboHook.md">here</a>.

CubesterYT commented:
Ohh ok lemme fix the errors real quick.

CubesterYT commented:
@GarboMuffin I fixed the lint issues, try Validation process again.

CubesterYT commented:
Yep I checked everything, all should be ready for merging now.

GarboMuffin commented on extensions/CubesterYT/ScratchHook.js:
```diff
@@ -0,0 +1,168 @@
+(function (Scratch) {
+  "use strict";
+
+  const icon = "https://raw.githubusercontent.com/CubesterYT/dependencies/main/TurboWarp%20Extensions/ScratchHook/Logo.png";
```
icon needs to be a data: URI

CubesterYT replied:
Ok I'll fix that

CubesterYT replied:
data URI now used.

GarboMuffin commented:
we try to avoid abusing Scratch trademarks, so ideally this would be called something other than "ScratchHook" and not use the Scratch logos at all

GarboMuffin commented on website/index.ejs:
```diff
@@ -518,6 +518,12 @@
+        <div class="extension">
+          <%- banner('CubesterYT/ScratchHook') %>
+          <h3>ScratchHook</h3>
+          <p>A fully modular extension that allows you to use Webhooks, with documentation <a href="https://github.com/CubesterYT/dependencies/blob/main/TurboWarp%20Extensions/ScratchHook/ScratchHook.md">here</a>. Created by <a href="https://scratch.mit.edu/users/CubesterYT/">CubesterYT</a>.</p>
```
if you have documentation then set a docsURI https://docs.turbowarp.org/development/extensions/assorted-apis#docsuri

GarboMuffin replied:
usually we want these to be hosted on extensions.turbowarp.org though, so you could put an HTML file in website/CubesterYT/ScratchHook.html and then use https://extensions.turbowarp.org/CubesterYT/ScratcHook.html

CubesterYT replied:
Is it alright for me to have it hosted on my own for now? I'll add the html documentation version soon.

CubesterYT replied:
HTML Docs added

CubesterYT commented:
> we try to avoid abusing Scratch trademarks, so ideally this would be called something other than "ScratchHook" and not use the Scratch logos at all

Can we have it in this case only?

GarboMuffin commented on extensions/CubesterYT/ScratchHook.js:
```diff
@@ -0,0 +1,168 @@
+    }
+
+    webhook ({hookDATA,hookURL}) {
+      const arrayDATA = hookDATA.split("{,}");
```
something tells me it should be possible to shorten a lot of this down

CubesterYT replied:
Since this is my first time dealing with Extensions, I can't figure out anything else, but this is fully functioning.

CubesterYT replied:
Due to the fact that it allows for users to not be restricted to a certain order when making webhook messages

CubesterYT commented:
@GarboMuffin Updated the Extension to change to data URI icon and have documentation in the actual extension itself.

CubesterYT commented:
I will add an html page some time after the merge, as I have tests this week. So in the next ScratchHook update, along side embed support, it should have itself an html documentation page.

CubesterYT commented:
If I can, I might be able to make the html page soon enough this week.

CubesterYT commented:
@GarboMuffin I have so far, as per the review, changed the icon to a data URI, added the documentation using docsURI and made a full html page for extensions.turbowarp.org. The extension is fully completed as of now, so it should be ready to be merged now.

CubesterYT commented:
@GarboMuffin The PR is fully done, the extension is prefected.

CubesterYT commented:
I hope it can be merged in time to be added to TurboWar Desktop 1.8

GarboMuffin commented:
I changed a good few things. if anything is broken let me know *as soon as possible*
